source/common/memory/stats.cc

```diff
@@ -45,10 +45,12 @@ uint64_t Stats::totalPageHeapUnmapped() {
 
 #else
 
+#include "common/memory/debug.h"
+
 namespace Envoy {
 namespace Memory {
 
-uint64_t Stats::totalCurrentlyAllocated() { return 0; }
+uint64_t Stats::totalCurrentlyAllocated() { return Debug::bytesUsed(); }
```

Comment:
I'd slightly prefer to have this whole block duplicated, so that we have:
```
#ifdef TCMALLOC
    ...
#elif defined(MEMORY_DEBUG_ENABLED)
    ...
#else 
    ...
#endif
```
instead of adding code to the build without debug memory.